src/function-builder.ts

```diff
@@ -44,45 +45,79 @@ import * as storage from './providers/storage';
 /**
  * Assert that the runtime options passed in are valid.
  * @param runtimeOptions object containing memory and timeout information.
- * @throws { Error } Memory and TimeoutSeconds values must be valid.
+ * @throws { Error } FailurePolicy, Memory and TimeoutSeconds values must be
+ *     valid.
  */
-function assertRuntimeOptionsValid(runtimeOptions: RuntimeOptions): boolean {
-  if (
-    runtimeOptions.memory &&
-    !_.includes(VALID_MEMORY_OPTIONS, runtimeOptions.memory)
-  ) {
-    throw new Error(
-      `The only valid memory allocation values are: ${VALID_MEMORY_OPTIONS.join(
-        ', '
-      )}`
-    );
+function assertRuntimeOptionsValidity(runtimeOptions: RuntimeOptions): void {
```

Comment:
I like the renaming and changing to void here, as well as all the other additional checks, thanks! Looks very clean.